### PR TITLE
Lab3

### PR DESCRIPTION
**Pull request by khadai**

### CODE REVIEW

**Comment by khadai** [comment]
not all requirements are met!

File: app/src/main/java/com/example/fisrtapplication/activities/ItemDetailsActivity.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.example.fisrtapplication.R;
+import com.squareup.picasso.Picasso;
+
+import java.util.Objects;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+public class ItemDetailsActivity extends AppCompatActivity {
+    private final int TARGET_WIDTH = 100;
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/activities/ItemDetailsActivity.java]: Must be `static`

File: app/src/main/java/com/example/fisrtapplication/activities/ItemDetailsActivity.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            String vendingGoods = getIntent().getStringExtra("vending_goods");
+            String vendingAddress = getIntent().getStringExtra("vending_address");
+            String vendingCompany = getIntent().getStringExtra("vending_company");
+            String imageName = getIntent().getStringExtra("vending_img_url");
+
+            setImage(vendingName, vendingGoods, vendingAddress, vendingCompany, imageName);
+        }
+    }
+
+    private void setImage(String vendingName, String vendingGoods, String vendingAddress, String vendingCompany, String imageName) {
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/activities/ItemDetailsActivity.java]: You set more than image here but method name does not reflect that.

File: app/src/main/java/com/example/fisrtapplication/activities/ItemDetailsActivity.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_item_details);
+
+        Objects.requireNonNull(getSupportActionBar()).setTitle("Vending Details");
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        getIncomingIntent();
+    }
+
+    private void getIncomingIntent() {
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/activities/ItemDetailsActivity.java]: displayVendingFromIntent

File: app/src/main/java/com/example/fisrtapplication/activities/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+public class MainActivity extends AppCompatActivity {
+    private FirebaseAuth mAuth;
+    private ProgressBar progressBar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Set the content of the activity to use the activity_main.xml layout file
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/activities/MainActivity.java]: These comments are useless. Need to be removed

File: app/src/main/java/com/example/fisrtapplication/adapters/CategoryAdapter.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    @NonNull
+    @Override
+    public Fragment getItem(int position) {
+        if (position == 0) {
+            return new DataListFragment();
+        } else if (position == 1) {
+            return new BlankFragment();
+        } else {
+            return new ProfileFragment();
+        }
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/adapters/CategoryAdapter.java]: Use `switch` operator instead

**Review by MarkoKuspis — COMMENTED** [review]
1. Lots of blank lines
2. Lots of copy-pasted and autogenerated code
3. Bad methods naming

File: app/src/main/java/com/example/fisrtapplication/adapters/VendingsAdapter.java [hunk 6]
```diff
@@ -1,27 +1,32 @@
...
 
 import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
 import androidx.recyclerview.widget.RecyclerView;
 
 public class VendingsAdapter extends RecyclerView.Adapter<VendingsAdapter.VendingViewHolder> {
-    private List<Vending> vendings;
-    private final int TARGET_WIDTH = 88;
-    private final int TARGET_HEIGHT = 86;
+    private final List<Vending> vendings;
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/adapters/VendingsAdapter.java]: `static` vars must always go first

File: app/src/main/java/com/example/fisrtapplication/fragments/DataListFragment.java [hunk 7]
```diff
@@ -1,15 +1,18 @@
-package com.example.fisrtapplication.activities;
+package com.example.fisrtapplication.fragments;
+
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/fragments/DataListFragment.java:2]: redundant blank line

File: app/src/main/java/com/example/fisrtapplication/fragments/DataListFragment.java [hunk 8]
```diff
@@ -18,76 +21,73 @@
...
-public class DataListActivity extends AppCompatActivity {
 
+public class DataListFragment extends Fragment {
     private RecyclerView recyclerView;
     private VendingsAdapter vendingsAdapter;
     private SwipeRefreshLayout refreshLayout;
     private FirebaseAuth mAuth;
     private ProgressBar progressBar;
     private View content;
+    private View rootView;
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/fragments/DataListFragment.java]: Please get rid of it. You can either pass view to your `setupViews()` method or use `onViewCreated()` + `getView()` instead

File: app/src/main/java/com/example/fisrtapplication/fragments/ProfileFragment.java [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+        startActivityForResult(intent, IMAGE_REQUEST);
+    }
+
+    private String getFileExtension(Uri uri) {
+        ContentResolver contentResolver = Objects.requireNonNull(getContext()).getContentResolver();
+        MimeTypeMap mimeTypeMap = MimeTypeMap.getSingleton();
+        return mimeTypeMap.getExtensionFromMimeType(contentResolver.getType(uri));
+    }
+
+    private void uploadImage() {
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/fragments/ProfileFragment.java:172]: This needs to be refactored, complexity is too high and it becomes incomprehensible.

File: app/src/main/java/com/example/fisrtapplication/utils/ApplicationEx.java [hunk 10]
```diff
@@ -21,10 +21,13 @@ public void onCreate() {
         vendingApiClient = createVendingApiClient();
     }
 
+
+
     public FirebaseAuth getAuth() {
         return mAuth;
     }
 
+
```
Comment (MarkoKuspis) [inline app/src/main/java/com/example/fisrtapplication/utils/ApplicationEx.java]: Seems like an invalid change to me

**Pull request merged**